app/api/v1/users.rb

```diff
@@ -0,0 +1,24 @@
+module V1
+  class Users < API
+    resources :users do
+      desc 'Return all users'
+      get do
+        @users = User.all
```

Comment:
you can merge main to your branch and use pundit here to restrict access to this endpoint
https://stackoverflow.com/a/31064629